src/api/CortexClient.ts

```diff
@@ -381,18 +386,27 @@ export class CortexClient implements CortexApi {
     input: RequestInfo,
     init?: RequestInit,
   ): Promise<Response> {
-    let token: string | undefined = undefined;
+    let token: string | undefined;
+    let email: string | undefined;
+    let displayName: string | undefined;
     if (this.identityApi !== undefined) {
       ({ token } = await this.identityApi.getCredentials());
+      const profileInfo = await this.identityApi.getProfileInfo();
```

Comment:
We should be able to fetch this in parallel with credentials, right?
```ts
const [{ token}, profileInfo] = await Promise.all(
  this.identityApi.getCredentials(),
  this.identityApi.getProfileInfo(),
);
``

Reply:
i can do the following, but destructuring the token in the assignment didn't work
```
      const [credentials, profileInfo] = await Promise.all([
        this.identityApi.getCredentials(),
        this.identityApi.getProfileInfo(),
      ]);
      token = credentials.token;
```

Reply:
Sure, that's fine. The important part is that the HTTP requests are made in parallel instead of sequence, which is accomplished by the `await Promise.all([])`